code/game/machinery/turrets.dm

```diff
@@ -804,6 +836,12 @@
 		sleep(2)
 	return
 
+/obj/structure/turret/gun_turret/proc/choose_projectile()
+	var/chosen_projectile = pick(available_projectiles)
+	for(var/I in restricted_projectiles)
+		if(chosen_projectile == I)
+			choose_projectile()
```

Comment:
You can avoid the need for recursion (slow, dangerous!) by picking from a list that has no restricted projectiles like so:
```suggestion
	var/chosen_projectile = pick(available_projectiles - restricted_projectiles)
```
At this point you might as well do away with the proc entirely.

Reply:
That's... comically simple, thanks
I just copied the roulette revolver implementation directly so that could use an update too lmao
Will update in next commit